traffic_ops/testing/api/v4/coordinates_test.go

```diff
@@ -198,3 +238,174 @@ func DeleteTestCoordinates(t *testing.T) {
+			}
+		} else {
+			t.Error("Error in getting coordinates by limit")
+		}
+
+		if len(coordinates) > 1 {
+			qparams = url.Values{}
+			qparams.Set("orderby", "id")
+			qparams.Set("limit", "1")
+			qparams.Set("offset", "1")
+			coordinatesWithOffset, _, err := TOSession.GetCoordinates(qparams, nil)
+			if err == nil {
+				if !reflect.DeepEqual(coordinates[1:2], coordinatesWithOffset) {
+					t.Error("expected GET Coordinates with limit = 1, offset = 1 to return second result")
+				}
+			} else {
+				t.Error("Error in getting coordinates by limit and offset")
+			}
+
+			qparams = url.Values{}
+			qparams.Set("orderby", "id")
+			qparams.Set("limit", "1")
+			qparams.Set("page", "2")
+			coordinatesWithPage, _, err := TOSession.GetCoordinates(qparams, nil)
+			if err == nil {
+				if !reflect.DeepEqual(coordinates[1:2], coordinatesWithPage) {
+					t.Error("expected GET Coordinates with limit = 1, page = 2 to return second result")
+				}
+			} else {
+				t.Error("Error in getting coordinates by limit and page")
+			}
+		} else {
+			t.Errorf("only one Coordinates found, so offset functionality can't test")
+		}
+	} else {
+		t.Errorf("No Coordinates found to check pagination")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "-2")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when limit is not bigger than -1")
+	} else if !strings.Contains(err.Error(), "must be bigger than -1") {
+		t.Errorf("expected GET Coordinates to return an error for limit is not bigger than -1, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("offset", "0")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when offset is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Coordinates to return an error for offset is not a positive integer, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("page", "0")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when page is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Coordinates to return an error for page is not a positive integer, actual error: " + err.Error())
+	}
+}
+
+func CreateTestCoordinatesWithInvalidName(t *testing.T) {
+	firstCoordinates := testData.Coordinates[0]
+	firstCoordinates.Name = ""
+	_, reqInf, err := TOSession.CreateCoordinate(firstCoordinates)
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+	if err == nil {
+		t.Errorf("Getting Coordinates by Invalid Name %v", err)
```

Comment:
this will always print `Getting Coordinates by Invalid Name <nil>`, which I think is confusing because it appears to be complaining about an error that's `nil` when actually we expected an error and didn't get one.

Reply:
updated the error response.